In [70]:
import pandas as pd
import numpy as np
import scipy.stats as st

In [2]:
master=pd.read_csv("PSID (1).csv")
master.shape

(4856, 9)

# Data Pre Processing

In [10]:
master1=master[master['kids']<98]
master1.shape
master2=master1[master1['educatn']<98]
master2.shape
master2['Kids_New']=np.where(master2['kids']<=2,"k1","k2")
master2['Married_New']=np.where(master2['married']=="married","m1","m2")
master2.shape

C:\Users\Heshan Perera\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """
C:\Users\Heshan Perera\Anaconda3\lib\site-packages\ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


(4528, 11)

# T test to see if there is a difference between earnings of married and other people

In [90]:
## Import the packages
import numpy as np
from scipy import stats
from scipy.stats import ttest_ind
master3=master2[master2['earnings']>0]
m1=master3[master3['Married_New']=='m1'].earnings
m2=master3[master3['Married_New']=='m2'].earnings

stat, p = ttest_ind(m1,m2)
p

0.027515538131075167

In [91]:
np.mean(m1)

19389.036617262424

In [93]:
st.t.interval(0.95, len(m1)-1, loc=np.mean(m1), scale=st.sem(m1))

(18720.20051509543, 20057.87271942942)

In [92]:
np.mean(m2)

18128.72017167382

In [94]:
st.t.interval(0.95, len(m2)-1, loc=np.mean(m2), scale=st.sem(m2))

(17268.555759225124, 18988.884584122516)

H0: Mariied and Other Peoples Average Earnings are equal
H1: Mariied and Other Peoples Average Earnings are not equal

Acccording to the t test pvalue is less than 0.05 that means we have to reject Ho at 95% level of confidence.
 
It concludes that average earnings of married and other people is not eaqual and by looking at the summary stats we noticed 
that average earnings of married people (19389) is higher than average earrnmings of other people 18128


95% confidence interval for Average earnings(per year) for married people (18720.20051509543, 20057.87271942942)

95% confidence interval for Average earnings(per year) other people (17268.555759225124, 18988.884584122516)
 




# T test to check whether people with age over 40 years earn more than the people with age less than 40 years

In [84]:
master3=master[master['earnings']>0]
master3.shape
master3['Ag_Cat']=np.where(master3['age']<40,"less than 40","over 40")



C:\Users\Heshan Perera\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [85]:
m1=master3[master3['Ag_Cat']=='less than 40'].earnings
m2=master3[master3['Ag_Cat']=='over 40'].earnings

stat, p = ttest_ind(m1,m2)
p

2.5060497396711643e-06

In [86]:
np.mean(m1)

17901.47563261481

In [87]:
st.t.interval(0.95, len(m1)-1, loc=np.mean(m1), scale=st.sem(m1))

(17232.00693238186, 18570.944332847757)

In [88]:
np.mean(m2)

20401.56324110672

In [89]:
st.t.interval(0.95, len(m2)-1, loc=np.mean(m2), scale=st.sem(m2))

(19604.987493851746, 21198.138988361694)

H0: Average earnings of a person with age less than 40 is equal to Average earning of a person with age greater than or equal to 40 years

H1:Average earnings of a person with age less than 40 is not equal to Average earning of a person with age greater than or equal to 40 years

Since the p value of the T test is less than 0.05 we can reject H0 with a 95% level of confidence

According to the sample Average earnings of a person with age less than 40 is equal to 17901 According to the sample Average earnings of a person with age greater than or equal to 40 is equal to 20401. Which shows that people with age over 40 years earns more than people with age less than 40

95% confidence interval for Average earnings(per year) of a person with age age greater than or equal to 40 years (17232.00693238186, 18570.944332847757)

95% confidence interval for Average earnings(per year) of a person with Low Educational Level (19604.987493851746, 21198.138988361694)



# T test to check whether people with age over 40 years works more than the people with age less than 40 years

In [77]:
master3=master[master['hours']>0]
master3.shape
master3['Ag_Cat']=np.where(master3['age']<40,"less than 40","over 40")


C:\Users\Heshan Perera\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [78]:
m1=master3[master3['Ag_Cat']=='less than 40'].hours
m2=master3[master3['Ag_Cat']=='over 40'].hours

stat, p = ttest_ind(m1,m2)
p

2.8967779055610274e-05

In [79]:
np.mean(m1)

1593.8561419897244

In [80]:
st.t.interval(0.95, len(m1)-1, loc=np.mean(m1), scale=st.sem(m1))

(1562.4515770097912, 1625.2607069696576)

In [81]:
np.mean(m2)

1695.9606557377049

In [82]:
st.t.interval(0.95, len(m2)-1, loc=np.mean(m2), scale=st.sem(m2))

(1660.3527598459957, 1731.568551629414)

H0: Average working hours(per year) of a person with age less than 40 is equal to Average working hours of a person with age greater than or equal to 40 years 

H1:Average working hours(per year) of a person with age less than 40 is not equal to Average working hours(per year) of a person with age greater than or equal to 40 years

Since the p value of the T test is less than 0.05 we can reject H0 with a 95% level of confidence

According to the sample Average working hours(per year) of a person with age less than 40 is equal to 1593 According to the sample Average working hours(per year) of a person with age greater than or equal to 40 is equal to 1695. Which shows that people with age over 40 years works more hours per year than people with age less than 40

95% confidence interval for Average working hours(per year) of a person with age less than 40 (1562.4515770097912, 1625.2607069696576)

95% confidence interval for Average working hours(per year) of a person person with age greater than or equal to 40 years (1660.3527598459957, 1731.568551629414)

# T test to check whether the earnings of people with high educational levels  are different  than  earnings of the people with low educational levels

In [43]:
master3=master[master['educatn']<98]
master4=master3[master3['earnings']>0]
master4.shape
master4['Ed_Cat']=np.where(master4['educatn']<14,"low","High")

C:\Users\Heshan Perera\Anaconda3\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  after removing the cwd from sys.path.


In [71]:
m1=master4[master4['Ed_Cat']=='low'].earnings
m2=master4[master4['Ed_Cat']=='High'].earnings

stat, p = ttest_ind(m1,m2)
p

7.713331104329767e-95

In [72]:
np.mean(m1)

14901.589937666964

In [73]:
st.t.interval(0.95, len(m1)-1, loc=np.mean(m1), scale=st.sem(m1))

(14433.644954567146, 15369.534920766782)

In [74]:
np.mean(m2)

26087.509508716324

In [75]:
st.t.interval(0.95, len(m2)-1, loc=np.mean(m2), scale=st.sem(m2))

(24994.934095784818, 27180.08492164783)

H0: Average earnings of a person with High Educational Level is equal to Average earning of a person with Low Educational Level

H1:Average earnings of a person with High Educational Level is not equal to Average earning of a person with Low Educational Level

Since the p value of the T test is less than 0.05 we can reject H0 with a 95% level of confidence

According to the sample Average earnings of a person High Educational Level(14 to 17) is equal to 26087 According to the sample Average earnings of a person with Low Educational Level(less than 14) is equal to 14901. Which shows that people with High Educational levels earns more than the people with low educational levels

95% confidence interval for Average earnings(per year) of a person with High Educational Level ((24994.934095784818, 27180.08492164783)

95% confidence interval for Average earnings(per year) of a person with Low Educational Level (14433.644954567146, 15369.534920766782)



# T test to check whether working hours of people with high educational levels  are different than the working hours of people with low educational levels

In [51]:
m1=master4[master4['Ed_Cat']=='low'].hours
m2=master4[master4['Ed_Cat']=='High'].hours

stat, p = ttest_ind(m1,m2)
p

0.09417055559257596

In [58]:
stats.describe(m1)

DescribeResult(nobs=2246, minmax=(6, 5160), mean=1620.3490650044523, variance=539175.0638454505, skewness=-0.24914196094497612, kurtosis=0.9774515488003894)

In [68]:
st.t.interval(0.95, len(m1)-1, loc=np.mean(m1), scale=st.sem(m1))

(1589.9652488287225, 1650.7328811801822)

In [60]:
stats.describe(m2)

DescribeResult(nobs=1262, minmax=(7, 4821), mean=1663.1584786053884, variance=508825.6449677073, skewness=-0.18024880550149047, kurtosis=0.7264489197730679)

In [69]:
st.t.interval(0.95, len(m2)-1, loc=np.mean(m2), scale=st.sem(m2))

(1623.7653967771334, 1702.5515604336433)

H0: Average working hours(per year) of a person with High Educational Level is equal to Average working hours of a person with Low Educational Level

H1:Average working hours(per year) of a person with High Educational Level is not equal to Average working hours(per year) of a person with Low Educational Level

Since the p value of the T test is not less than 0.05 we do not reject H0 with a 95% level of confidence

According to the sample Average working hours(per year) of a person with High Educational is equal to 1663 According to the sample Average working hours(per year) of a person with Low Educational is equal to 1620. This shows that there is not significant difference between working hours of a person with high educational level and a person with a low educational level.

95% confidence interval for Average working hours(per year) of a person with High Educational Level
(1623.7653967771334, 1702.5515604336433)

95% confidence interval for Average working hours(per year) of a person with Low Educational Level
(1589.9652488287225, 1650.7328811801822)

